### PR TITLE
Add /ok route as an express middleware

### LINKED ISSUE #4375: OK page for Volto
**Is your feature request related to a problem? Please describe.**
In Plone we currently have the `@@ok` view where Plone returns a simple `OK` message to signal that it is alive and thus this URL can be used in load-balancers or keep-alive services to check that the website is running.

We do not have such a feature in Volto and in some cases if the back-end is not exposed the only way to check if the website is alive is to hit the front-page or some other pages.

**Describe the solution you'd like**
I would like to have a non content route called `ok` in Volto, that renders a component that simply returns `OK` like the Plone one does. This way this URL can be added to any service checking the status of the site.

### PR DESCRIPTION
**Pull request by erral**
Fixes #4375 adding an Express middleware

This supersedes #4429

### CODE REVIEW

**Review by tiberiuichim — APPROVED** [review]

**Review by tiberiuichim — CHANGES_REQUESTED** [review]
In Plone the ok view is namespaced by the `@@` symbols, so there's no chance to overlap something else. Let's do something similar here. Maybe `/_ok` ? 

Also, I don't think there's a need for a setting for this, if we solve the namespacing problem.

@plone/volto-team thoughts on this?

**Comment by davisagli** [comment]
How about something like `.well-known/volto/ok` (based on https://en.wikipedia.org/wiki/Well-known_URI)?

Whatever it is, it should be mentioned in the docs.

**Comment by avoinea** [comment]
> In Plone the ok view is namespaced by the `@@` symbols, so there's no chance to overlap something else. Let's do something similar here. Maybe `/_ok` ?
> 
> Also, I don't think there's a need for a setting for this, if we solve the namespacing problem.
> 
> @plone/volto-team thoughts on this?

Well, it also works with `/ok` in Plone Classic and we're doing all health-checks to this path `/ok`. Thus, I would stick to `/ok`. Or use the `.well-known` approach as @davisagli suggested.

**Comment by sneridagh** [comment]
We need a consensus here. I'm ok with using both `/ok` and `.well-known/volto/ok` the reasoning behind both seems ok (no pun intended) for me. Everybody is ok? 😂

**Comment by tiberiuichim** [comment]
I'm +1 on adopting the industry standard.

**Comment by erral** [comment]
I don't see a ok URL under the well-known URLs right? I mean I don't mind using the .well-known approach, but right know there's no such standard for ok right?

We can start having the ok like we have in classic and evolve to whatever is stablished later on.

**Comment by davisagli** [comment]
I was the one who brought up the `.well-known` idea, but I don't see any big problem with `/ok`: it is unlikely to conflict with content, and if it does there is the setting. So I would go ahead with this the way it is. The one thing I am missing in the PR is documentation of the new setting.

**Comment by stevepiercy** [comment]
> The one thing I am missing in the PR is documentation of the new setting.

Would that go here?

https://6.docs.plone.org/volto/configuration/settings-reference.html#server-specific-serverconfig

**Comment by erral** [comment]
I have realized that I was missing the documentation part on this PR as requested by @stevepiercy :facepalm: so I have added it :wink:

File: docs/source/configuration/settings-reference.md [hunk 1]
```diff
@@ -374,8 +374,17 @@ additionalToolbarComponents
       }}
     </Plug>
     ```
-```
+````
```
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md]: This is the closing fence for the opening glossary fence at the start of the page. It should be moved after your new glossary entry.

```suggestion
```

File: docs/source/configuration/settings-reference.md [hunk 2]
```diff
@@ -374,8 +374,17 @@ additionalToolbarComponents
...
+
+okView
+Volto provides a `/ok` URL where it respondes with a text/plain ok response (with an `HTTP 200` status code) to signal third-party health-check services that the Volto process is running correctly.
```
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md]: Indent glossary definitions.

```suggestion
    Volto provides an `/ok` URL where it responds with a `text/plain ok` response, with an `HTTP 200` status code, to signal third party health check services that the Volto process is running correctly.
```

File: docs/source/configuration/settings-reference.md [hunk 3]
```diff
@@ -374,8 +374,17 @@ additionalToolbarComponents
...
+Using this setting one can modify such an URL and configure to respond in some other URL.
```
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md]:
```suggestion
    Using this setting, one can modify such an URL and configure it to respond with another URL.
```

File: docs/source/configuration/settings-reference.md [hunk 4]
```diff
@@ -374,8 +374,17 @@ additionalToolbarComponents
...
+The provided default matches the existing Plone Classic URL.
```
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md]:
```suggestion
    The provided default matches the existing Plone Classic UI URL.
```

File: docs/source/configuration/settings-reference.md [hunk 5]
```diff
@@ -374,8 +374,17 @@ additionalToolbarComponents
...
 
+    ```jsx
+      config.settings.okView = '/site-is-ok'
+
```
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:394]:
```suggestion
```

File: docs/source/configuration/settings-reference.md [hunk 6]
```diff
@@ -374,8 +374,17 @@ additionalToolbarComponents
...
+    ```
```
Comment (stevepiercy) [inline docs/source/configuration/settings-reference.md:395]: Close glossary fence.

````suggestion
    ```
```
````

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Thanks for the docs. Mostly MyST and English syntax and grammar.

**Comment by erral** [comment]
> Thanks for the docs. Mostly MyST and English syntax and grammar.

thank you as always!

**Review by stevepiercy — APPROVED** [review]
Looks good! https://deploy-preview-4432--volto.netlify.app/configuration/settings-reference.html#term-okRoute

**Comment by stevepiercy** [comment]
It is OK to ignore the failing docs job. He Who Shall Not Be Named has once again pulled more feathers from the bird site, now requiring authentication to view a user's twerks.

**Review by sneridagh — APPROVED** [review]

**Pull request merged**